onmt/Seq2Seq.lua

```diff
@@ -85,12 +87,32 @@ function Seq2Seq:getOutput(batch)
   return batch.targetOutput
 end
 
+function Seq2Seq:maskPadding(batch)
+  self.models.encoder:maskPadding()
+  if self.args.uneven_batches then
+    if batch.uneven then
+      self.models.decoder:maskPadding(batch.sourceSize, batch.sourceLength)
+    else
+      self.models.decoder:maskPadding()
+    end
+  end
+end
+
 function Seq2Seq:forwardComputeLoss(batch)
+  self:maskPadding(batch)
   local encoderStates, context = self.models.encoder:forward(batch)
   return self.models.decoder:computeLoss(batch, encoderStates, context, self.criterion)
 end
 
 function Seq2Seq:trainNetwork(batch, dryRun)
+  if self.args.uneven_batches then
```

Comment:
Let just call `self:maskPadding(batch)` to avoid duplication even if `self.models.encoder:maskPadding()` does nothing on a second call.